api/src/controllers/login.js

```diff
@@ -51,6 +62,31 @@ const templates = {
       'privacy.policy'
     ],
   },
+  passwordReset: {
+    file: path.join(__dirname, '..', 'templates', 'login', 'password-reset.html'),
+    translationStrings: [
+      'login.show_password',
+      'login.hide_password',
+      'change.password.title',
+      'change.password.hint',
+      'change.password.submit',
+      'change.password.new.password',
+      'change.password.confirm.password',
+      'password.weak',
+      'password.length.minimum',
+      'password.must.match',
+      'user.password.current',
+      'password.current.incorrect',
+      'password.same'
+    ],
+  }
+};
+
+const skipPasswordChange = async (user, userCtx) => {
+  return !user?.password_change_required ||
+    user?.token_login?.active ||
+    roles.isDbAdmin(userCtx) ||
+    await auth.hasAllPermissions(userCtx, 'can_skip_password_change');
```

Comment:
thought: referencing back to my [previous comment](https://github.com/medic/cht-core/pull/9581#discussion_r1880794413) about how this feature feels like a bit of a breaking change to have enabled by default, I initially expected this behavior to be something _toggled ON by a permission_.  E.g. the system would trigger the reset password flow only for users with the `can_reset_password_assigned_by_admin` permission, or something like that...

Reply:
Great observation, I initially thought that as well. Even started working on it from that approach. However, Gareth raised a fair point. Our purpose for this feature is to improve security to safe guard against a password leak. 

If we leave it as an opt in feature, because people are generally "lazy" when it comes to security steps, a project will not turn it on. Therefore, in the event of a password leak, the feature won't really help them. 

But having it as an opt out then that way the system is designed to safe guard, but you can opt out knowingly at your own risk.